bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -258,13 +268,52 @@ public void filter(@Nullable ContainerRequestContext requestContext) throws IOEx
                             }
                         }
                     }
-                } else if (implicitUserRole) {
+                } else if (isImplicitUserRole(requestContext)) {
                     requestContext.setSecurityContext(new AnonymousUserSecurityContext());
                 }
             } catch (AuthenticationException e) {
-                logger.warn("Unauthorized API request from {}: {}", servletRequest.getRemoteAddr(), e.getMessage());
+                logger.warn("Unauthorized API request from {}: {}", getClientIp(requestContext), e.getMessage());
                 requestContext.abortWith(JSONResponse.createErrorResponse(Status.UNAUTHORIZED, "Invalid credentials"));
             }
         }
     }
+
+    private boolean isImplicitUserRole(ContainerRequestContext requestContext) {
+        if (implicitUserRole) {
+            return true;
+        }
+        String clientIp = getClientIp(requestContext);
+        return trustedNetworks.stream().anyMatch(networkCIDR -> isIpInSubnet(clientIp, networkCIDR));
+    }
+
+    private String getClientIp(ContainerRequestContext requestContext) {
+        String ipForwarded = Objects.requireNonNullElse(requestContext.getHeaderString("x-forwarded-for"), "");
+        String clientIp = ipForwarded.split(",")[0];
+        return clientIp.isBlank() ? servletRequest.getRemoteAddr() : clientIp;
+    }
+
+    private boolean isIpInSubnet(final String ip, final String subnetCDIR) {
```

Comment:
I think this code should be improved as it does a lot of things that can be pre-parsed. If you do something like

```
private static class CIDR {
    private byte[] networkBytes;
    private long prefix;

    public CIDR(String cidr) {
        String[] parts = cidr.split("/");
        this.prefix = Integer.parseInt(parts[1]);
        this.bytes =  InetAddress.getByName(parts[0]).getAddress();
    }

    public boolean inRange(byte[] address) {
        <... enter your code here ...>
    }
}
```

and then do `byte[] address = InetAddress.getByName(ip).getAddress()` before l. 286, you only have to do the `String/byte[]` conversion only once and the parsing of the network parts only when parsing the config. This saves a ton of compute time for each request.

Reply:
Another option would be something along the lines of https://www.baeldung.com/java-check-ip-address-range#using-javas-inetaddress-class-for-ipv4, storing the range with start and end values and checking if it is between those. Of course we would need to use  `BigInteger` to make sure this also works for IPv6.